vnet/delay_filter.go

```diff
@@ -118,8 +118,7 @@ func (f *DelayFilter) processReadyPackets(now time.Time) {
 		if next == nil {
 			break
 		}
-
-		if chunk, ok := next.(timedChunk); ok && chunk.deadline.Before(now) {
+		if chunk, ok := next.(timedChunk); ok && (chunk.deadline.Equal(now) || chunk.deadline.Before(now)) {
```

Comment:
What do you think about?

```suggestion
		if chunk, ok := next.(timedChunk); ok && !chunk.deadline.After(now) {
```